File: LeetCode/136. Single Number/singleNumber.py
from typing import List 
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        seen = {}
        for num in nums: 
            seen[num] = seen.get(num, 0) + 1
        for num, count in seen.items():
            if count == 1:
                return num

solution = Solution()
print(solution.singleNumber([4,1,2,1,2]))

File: LeetCode/136. Single Number/singleNumber.ts
export function singleNumber(nums: number[]): number {
    const seen: { [key: number]: number } = {};
    for (const num of nums ) {
        seen[num] = (seen[num] || 0) + 1
    } 
    for (const [num, count] of Object.entries(seen)) {
        if (count === 1) {
            return Number(num);
        }
    }
    return 0;
}

console.log(singleNumber([4, 1, 2, 1, 2]))